packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -227,6 +239,12 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
 
   const onTextFieldKeyDown = useCallback(
     (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      /* @conditional-compile-remove(mention) */
+      // caretIndex should be set to undefined when the user is typing
+      setCaretIndex(undefined);
+      // shouldHandleOnMouseDownDuringSelect should be set to false after the last mouse down event
+      /* @conditional-compile-remove(mention) */
+      setShouldHandleOnMouseDownDuringSelect(false);
```

Comment:
Yes, it's needed. `onSelect` event can be called a couple of times for consequences of `onMouseDown` event (rerender, setSelectionRange) and `onMouseUp` can be called before or after `onSelect` calls as its order depends only on mouse events and not on select event. In this case, we can't use onMouseUp as a good place to update mouse events bool flags. Because of this, we need to ensure that when we have onKeyDown event, we clean up all variables from onMouseDown and are ready to handle keyboard event.